clang/test/3C/arrstructcallermulti2.c

```diff
@@ -110,8 +110,8 @@ static int *mul2(int *x) {
 }
 
 int * sus(struct general * x, struct general * y) {
-	//CHECK_NOALL: int * sus(struct general * x, _Ptr<struct general> y) {
-	//CHECK_ALL: _Array_ptr<int> sus(struct general * x, _Ptr<struct general> y) : count(5) {
+	//CHECK_NOALL: int *sus(struct general *x : itype(_Ptr<struct general>), _Ptr<struct general> y) : itype(_Ptr<int>) {
+	//CHECK_ALL: _Array_ptr<int> sus(struct general *x : itype(_Ptr<struct general>), _Ptr<struct general> y) : count(5) {
```

Comment:
I think at this point, I need to acknowledge before going too much further that my working knowledge of itypes may not be sophisticated enough to catch some potential bugs. 
In tests like this one, I'm willing to wholeheartedly agree with the itype insertion, because the name of the test indicates that only the caller uses things unsafely, and so it makes intuitive sense to me that we would give the callee itypes to indicate that the arguments are being used safely internally to this function. 
In other places (specifically pointers), I'm less sure of what it means to be correct, so it'd be worthwhile if someone could take 5 tests (that aren't autogenerated maybe) and give them a quick once-over.